### ISSUE TITLE
vagrant ansible 2.3.0 regression

### ISSUE THREAD

**Issue by shoreflyer** [mitchellh/vagrant#8391]
It seems vagrants ansible provisioner isn't ready for ansible's 2.3.0 release.

### Vagrant version
1.9.2

### Host operating system
macOS 10.12.3 (16D32)

### Guest operating system
ubuntu/xenial64        (virtualbox, 20170320.0.0)

### Vagrantfile
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

VAGRANTFILE_API_VERSION = "2"
MACHINE_HOSTNAME = "oceans"


$script = <<SCRIPT
apt-get install python nginx -y
SCRIPT

Vagrant.configure(VAGRANTFILE_API_VERSION) do |cfg|
  cfg.vm.box = "ubuntu/xenial64"
  cfg.vm.network "private_network", ip: "192.168.33.20"
  cfg.vm.synced_folder ".", "/vagrant"

  cfg.vm.provider :virtualbox do |vb|
    vb.customize ["modifyvm", :id, "--name", MACHINE_HOSTNAME]
    vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
    vb.customize ["modifyvm", :id, "--memory", 1024]
    vb.customize ["modifyvm", :id, "--cpus", 2]
    vb.customize ["modifyvm", :id, "--ioapic", "on"]
  end

  cfg.vm.provision "shell", inline: $script

  cfg.vm.provision "ansible" do |a|
    a.playbook = "site.yml"
    a.sudo = true
    a.extra_vars = { ansible_ssh_user: 'ubuntu' }
  end
end
```
### Debug output
https://gist.github.com/shoreflyer/9be5342bb4c569de913483dddd8d2ce5

### Expected behaviour
Completed provisioning or error within the ansible playbook.

### Actual behaviour
Errors out with the following:
```
...
==> default: Running provisioner: ansible...
    default: Running ansible-playbook...
 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [all] *********************************************************************
skipping: no hosts matched

PLAY RECAP *********************************************************************
...
```

**Comment by gildegoma** [comment]
@shoreflyer Thanks for this report. While trying to reproduce this issue, I hit a bug in Ansible 2.3 RC1 (see https://github.com/ansible/ansible/issues/22861). Although your debug output doesn't look the same, I suspect that your troubles are probably related to this. Which pre-release of Ansible 2.3 are you using?

I'm keeping the issue open for the moment to gather more information, but I hope everything will be fixed upstream before Ansible 2.3 goes GA. Thanks!

**Comment by shoreflyer** [comment]
@gildegoma you are right, I can reproduce the bug mentioned [ansible/ansible#22861](https://github.com/ansible/ansible/issues/22861).
It seems to be an ansible issue with 2.3.0 rc1. I'll keep an eye out for changes and update here accordingly but now I don't think it is really an issue with vagrant anymore.

Thanks!

**Comment by shoreflyer** [comment]
@gildegoma I've been able to recreate the error you had and updated the report output accordingly.
I can however run ansible 2.3.0 locally with no errors. Sectioned inventory or not.

**Comment by shoreflyer** [comment]
After updating to the latest ansible 2.3.0-stable everything is back to normal.
The fix for ansible/ansible#22861 resolved the problem.

Thanks
